hardware/raspberrypi/peripheral_addresses.md

```diff
@@ -1,32 +1,56 @@
 # Peripheral Addresses
 
-If there is no kernel driver available, and a program needs to access a peripheral address directly with mmap, it needs to know where in the virtual memory map the peripheral bus segment has been placed. This varies according to which model of Raspberry Pi is being used, so there are three helper function available to provide platform independence. **Note**: please use these functions rather than hardcoded values, as this will ensure future compatibility.
+If there is no kernel driver available, and a program needs to access a peripheral address directly with mmap, it needs to know where in the virtual memory map the peripheral bus segment has been placed. This varies according to which model of Raspberry Pi is being used, so there are three helper functions in [bcm_host.c](https://github.com/raspberrypi/userland/blob/3fd8527eefd8790b4e8393458efc5f94eb21a615/host_applications/linux/libs/bcm_host/bcm_host.c) to help provide platform independence.
```

Comment:
Ugh, please link to the `master` revision rather than a specific commit. And I wonder if it'd be better to link to the `.h` file rather than the `.c` ?

Reply:
> And I wonder if it'd be better to link to the .h file rather than the .c ?
I wondered about that. Personally, I mainly look at the source code to understand the implementation, so the `.c` file is more relevant to me. Either way is fine, and I figured that linking to sources is good for helping the reader understand in depth, but I wouldn't push the point if we don't like linking to source.

> Ugh, please link to the master revision rather than a specific commit.
Sorry, I haven't worked with github so much. If my pull requests come from my master branch, does that mean that separate fixes for separate issues will get all blended together?

Reply:
> Personally, I mainly look at the source code to understand the implementation

Your approach here is probably different to the majority of people reading this page? :wink:  Most people are probably just happy to use code, without understanding the low-level of how it actually works? (which is why I suggested that linking to the header might be more useful)

> > Ugh, please link to the master revision rather than a specific commit.

> If my pull requests come from my master branch...

You misunderstood my comment; I wasn't referring to this PR, but specifically to this URL:
`https://github.com/raspberrypi/userland/blob/3fd8527eefd8790b4e8393458efc5f94eb21a615/host_applications/linux/libs/bcm_host/bcm_host.c`

It would be better to link to `https://github.com/raspberrypi/userland/blob/master/host_applications/linux/libs/bcm_host/bcm_host.c`